Sources/Valkey/Cluster/ValkeyClusterClient.swift

```diff
@@ -159,21 +159,42 @@ public final class ValkeyClusterClient: Sendable {
 
         while true {
             do {
-                let respToken = try await self.retryingSend(
-                    clientSelector: clientSelector,
-                    command: command,
-                    logger: logger
-                )
-                return try Command.Response(fromRESP: respToken)
-            } catch let error as ValkeyClientError where error.errorCode == .commandError {
-                guard let errorMessage = error.message, let movedError = ValkeyMovedError(errorMessage) else {
-                    throw error
+                while !Task.isCancelled {
```

Comment:
This looks wyld. What does the outer `while true` and `do` do?

Reply:
Yeah It looks wrong, this has only been exposed because I merged `retryingSend` into `send`. I've collapsed it down to a single loop now.